Tasks/AzureFileCopy/Utility.ps1

```diff
@@ -984,190 +928,209 @@ function Get-AzureVMsCredentials
-                }
-                $Jobs.Remove($job.Id)
+            if($additionalArguments -eq "")
+            {
+                $additionalArguments = "/Z:`"$azCopyDestinationPath`" /S /Y"
             }
+
+            $azCopyCommand = "& `"$azCopyExeLocation`" /Source:$containerURL /Dest:`"$targetPath`" /SourceSAS:`"$containerSasToken`" $additionalArguments"
+            Invoke-Expression $azCopyCommand
+        }
+        finally
+        {
+            # Delete AzCopy tool folder
+            Get-ChildItem -Path $azCopyDestinationPath -Recurse -Force | Remove-Item -Force -Recurse -ErrorAction SilentlyContinue
+            Remove-Item $azCopyDestinationPath -Force -ErrorAction SilentlyContinue
+            Remove-Item $randomFolderPath -Force -ErrorAction SilentlyContinue
         }
     }
 
-    # While copying parallelly, if copy failed on one or more azure VMs then throw
-    if ($parallelOperationStatus -eq "Failed")
-    {
-        foreach ($error in $dtlsdkErrors) {
-            Write-Verbose "Error: $error"
-        }
+    $azCopyToolFileNamesString = $azCopyToolFileNames -join ";"
+    $azCopyToolFileContentsString = $azCopyToolFileContents -join ";"
 
-        Write-Telemetry "DTLSDK_Error" "CopyFilesParallellyToAzureVMsFailed"
-        $errorMessage = (Get-VstsLocString -Key "AFC_ParallelCopyFailed")      
-        ThrowError -errorMessage $errorMessage
+    # script block arguments
+    $scriptBlockArgs = " -containerURL '$containerURL' -targetPath '$targetPath' -containerSasToken '$containerSasToken' -additionalArguments '$additionalArguments' -azCopyToolFileNamesString '$azCopyToolFileNamesString' -azCopyToolFileContentsString '$azCopyToolFileContentsString'"
+    if($cleanTargetBeforeCopy)
+    {
+        $scriptBlockArgs += " -CleanTargetBeforeCopy"
     }
-}
 
-function Copy-FilesToAzureVMsFromStorageContainer
-{
-    param([string][Parameter(Mandatory=$true)]$storageAccountName,
-          [string][Parameter(Mandatory=$true)]$containerName,
-          [string][Parameter(Mandatory=$true)]$containerSasToken,
-		  [string]$blobStorageEndpoint,
-          [string][Parameter(Mandatory=$true)]$targetPath,
-          [string][Parameter(Mandatory=$true)]$azCopyLocation,
-          [string][Parameter(Mandatory=$true)]$resourceGroupName,
-          [object][Parameter(Mandatory=$true)]$azureVMResourcesProperties,
-          [object][Parameter(Mandatory=$true)]$azureVMsCredentials,
-          [string][Parameter(Mandatory=$true)]$cleanTargetBeforeCopy,
-          [string]$communicationProtocol,
-          [string]$skipCACheckOption,
-          [string][Parameter(Mandatory=$true)]$enableDetailedLoggingString,
-          [string]$additionalArguments,
-          [string][Parameter(Mandatory=$true)]$copyFilesInParallel,
-          [string][Parameter(Mandatory=$true)]$connectionType)
+    # other invoke-remotescript arguments
+    $sessionName = "AFCCopyToVMs"
```

Comment:
sessionName is used to reconnect to the target machines in case of any connection break. The target can have multiple sessions with this name. It will be better to have a guid here instead. Infact, now that I think about, it, sessionName should not have been exposed in the first place. I'll fix it. Do not pass sessionName anymore.

Reply:
done